tests/unit/test_metrics.py

```diff
@@ -0,0 +1,46 @@
+"""Testing functions of metrics module."""
+
+from datetime import datetime
+
+import numpy as np
+import polars as pl
+import pytest
+
+from wristpy.core.models import Measurement
+from wristpy.processing.metrics import euclidean_norm_min_one
+
+TEST_LENGTH = 100
+
+
+@pytest.mark.parametrize(
+    "x,y,z, expected_enmo",
+    [
+        (np.zeros(TEST_LENGTH), np.zeros(TEST_LENGTH), np.zeros(TEST_LENGTH), 0),
+        (np.ones(TEST_LENGTH), np.zeros(TEST_LENGTH), np.zeros(TEST_LENGTH), 0),
+        (np.ones(TEST_LENGTH) * 2, np.zeros(TEST_LENGTH), np.zeros(TEST_LENGTH), 1),
+        (np.ones(TEST_LENGTH), np.ones(TEST_LENGTH) * 2, np.ones(TEST_LENGTH) * 2, 2),
+    ],
+)
+def test_euclidean_norm_min_one(
+    x: np.ndarray, y: np.ndarray, z: np.ndarray, expected_enmo: float
+) -> None:
+    """Tests the euclidean norm function."""
+    dummy_date = datetime(2024, 5, 2)
+    dummy_datetime = pl.Series("time", [dummy_date] * TEST_LENGTH)
+    test_acceleration = Measurement(
+        measurements=np.column_stack((x, y, z)), time=dummy_datetime
+    )
+
+    enmo_results = euclidean_norm_min_one(test_acceleration)
+
+    assert np.all(
+        np.isclose(enmo_results.measurements, expected_enmo)
+    ), f"Expected {expected_enmo}"
```

Comment:
I know I initially used `np.isclose` but for these simple cases should we use `np.array_equal`?

Reply:
Never use `array_equal` (or `==`) on floating point values.

```Python
import numpy as np

arr1 = np.array([0.1]) + np.array([0.2])
arr2 = np.array([0.3])

print("arr1", arr1)
print("arr2", arr2)
print("equal", np.equal(arr1, arr2))
print("array_equal", np.array_equal(arr1, arr2))
print("isclose", np.isclose(arr1, arr2))
print("allclose", np.allclose(arr1, arr2))
```
```
arr1 [0.3]
arr2 [0.3]
equal [False]
array_equal False
isclose [ True]
allclose True
```